cmd2/history.py

```diff
@@ -222,5 +228,19 @@ def regex_search(self, regex: str) -> List[HistoryItem]:
 
         def isin(hi):
             """filter function for doing a regular expression search of history"""
-            return finder.search(hi) or finder.search(hi.expanded)
+            return finder.search(hi.statement.raw) or finder.search(hi.statement.expanded_command_line)
         return [itm for itm in self if isin(itm)]
+
+    def truncate(self, max_length: int) -> None:
+        """Truncate the length of the history, dropping the oldest items if necessary
+
+        :param max_length: the maximum length of the history, if negative, all history
+                           items will be deleted
+        :return: nothing
+        """
+        if max_length <= 0:
+            # remove all history
+            del self[:]
+        elif len(self) > max_length:
+            last_element = len(self) - max_length
+            del self[0:last_element]
```

Comment:
Would it be useful to add `__str__` and/or `__repr__` dunder methods to the `HistoryItem` class?

Reply:
Very useful. Done.